x-pack/plugins/lens/public/heatmap_visualization/visualization.tsx

```diff
@@ -0,0 +1,394 @@
+      </I18nProvider>,
+      domElement
+    );
+  },
+
+  toExpression(state, datasourceLayers, attributes): Ast | null {
+    const datasource = datasourceLayers[state.layerId];
+
+    const originalOrder = datasource.getTableSpec().map(({ columnId }) => columnId);
+    // When we add a column it could be empty, and therefore have no order
+
+    if (!originalOrder) {
+      return null;
+    }
+
+    return {
+      type: 'expression',
+      chain: [
+        {
+          type: 'function',
+          function: FUNCTION_NAME,
+          arguments: {
+            title: [attributes?.title ?? ''],
+            description: [attributes?.description ?? ''],
+            xAccessor: [state.xAccessor ?? ''],
+            yAccessor: [state.yAccessor ?? ''],
+            valueAccessor: [state.valueAccessor ?? ''],
+            legend: [
+              {
+                type: 'expression',
+                chain: [
+                  {
+                    type: 'function',
+                    function: LEGEND_FUNCTION,
+                    arguments: {
+                      isVisible: [state.legend.isVisible],
+                      position: [state.legend.position],
+                    },
+                  },
+                ],
+              },
+            ],
+            gridConfig: [
+              {
+                type: 'expression',
+                chain: [
+                  {
+                    type: 'function',
+                    function: HEATMAP_GRID_FUNCTION,
+                    arguments: {
+                      // grid
+                      strokeWidth: state.gridConfig.strokeWidth
+                        ? [state.gridConfig.strokeWidth]
+                        : [],
+                      strokeColor: state.gridConfig.strokeColor
+                        ? [state.gridConfig.strokeColor]
+                        : [],
+                      cellHeight: state.gridConfig.cellHeight ? [state.gridConfig.cellHeight] : [],
+                      cellWidth: state.gridConfig.cellWidth ? [state.gridConfig.cellWidth] : [],
+                      // cells
+                      isCellLabelVisible: [state.gridConfig.isCellLabelVisible],
+                      // Y-axis
+                      isYAxisLabelVisible: [state.gridConfig.isYAxisLabelVisible],
+                      yAxisLabelWidth: state.gridConfig.yAxisLabelWidth
+                        ? [state.gridConfig.yAxisLabelWidth]
+                        : [],
+                      yAxisLabelColor: state.gridConfig.yAxisLabelColor
+                        ? [state.gridConfig.yAxisLabelColor]
+                        : [],
+                      // X-axis
+                      isXAxisLabelVisible: [state.gridConfig.isXAxisLabelVisible],
+                    },
+                  },
+                ],
+              },
+            ],
+          },
+        },
+      ],
+    };
+  },
+
+  toPreviewExpression(state, datasourceLayers): Ast | null {
```

Comment:
I found the preview was not very easy to understand, maybe because there is no cell stroke. Have you tried it with cell stroke?

<img width="175" alt="Screen Shot 2021-05-14 at 3 26 46 PM" src="https://user-images.githubusercontent.com/666475/118321645-e0ee3380-b4cb-11eb-8690-db69a7f307b3.png">

Reply:
Updated in https://github.com/elastic/kibana/pull/97978/commits/b581f1c84376cc4eed287fbdf80463cf6a16a56a